### PR TITLE
Update mxo 1.2.0

### PR DESCRIPTION
**Pull request by chpypl**
Thank you for your contribution to Braintree. 

> Before submitting this PR, note that we cannot accept language translation PRs. We support the same languages that are supported by PayPal, and have a dedicated localization team to provide the translations. If there is an error in a specific translation, you may open an issue and we will escalate it to the localization team.

### Summary of changes

- Creating the auth config for android for merchant passed email

 ### Checklist

 - [x] Added a changelog entry

-

### CODE REVIEW

File: PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeCheckoutClient.java
Comment (scannillo) [inline PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeCheckoutClient.java:181]: We we OK to pass in a null `authConfig` if email isn't set?

**Review by scannillo — APPROVED** [review]
LGTM! Small comment on if passing in a null `authConfig` is ideal, but will defer to the MXO experts on that decision

File: PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeCheckoutClient.java
Comment (chpypl) [reply PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeCheckoutClient.java:181]: Yes we handle nullability in the SDK if authConfig is null

**Review by sshropshire — APPROVED** [review]

File: PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeRequest.java
Comment (sarahkoop) [inline PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeRequest.java:219]:
```suggestion
```

File: PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeRequest.java
Comment (jaxdesmarais) [inline PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeRequest.java:54]:
```suggestion
    private String userAuthenticationEmail;
    
```

File: PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeRequest.java
Comment (jaxdesmarais) [inline PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeRequest.java:216]:
```suggestion
    public void setUserAuthenticationEmail(@Nullable String userAuthenticationEmail) {
```

File: CHANGELOG.md
Comment (sarahkoop) [inline CHANGELOG.md:7]:
```suggestion
  * Added support for merchant passed email (opt in only)
```

File: CHANGELOG.md
Comment (sarahkoop) [inline CHANGELOG.md:8]:
```suggestion
```

File: CHANGELOG.md
Comment (jaxdesmarais) [inline CHANGELOG.md:7]: Can we update this to something like "add `setUserAuthenticationEmail() to PayPalNativeRequest"?

Comment (sarahkoop) [inline CHANGELOG.md:7]:
```suggestion
  * Add support for merchant passed email (opt in only)
```

**Review by sarahkoop — APPROVED** [review]
A few nitpick suggestions but LGTM

File: PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeCheckoutClient.java
Comment (sarahkoop) [inline PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeCheckoutClient.java:164]: Can we add a unit test for this portion of the logic?

Comment (chpypl) [reply PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeCheckoutClient.java:164]: As discussed we can disregard this.

File: PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeRequest.java
Comment (jaxdesmarais) [inline PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeRequest.java:54]: nit: I think we still want a newline after this

**Review by jaxdesmarais — APPROVED** [review]

**Pull request merged**
